dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -255,10 +259,10 @@ void Trace::resetPersistantCustomAppender_(const std::string& tag, SeverityLevel
   set.insert("Thread", attrs::make_attribute_value(currentId));
 
   TraceSinks& traceSink = sinks_.at(currentId);
-  for (vector< boost::shared_ptr<FileSink> >::const_iterator itSinks = traceSink.persistantSinks.begin();
+  for (std::unordered_map<std::string, boost::shared_ptr<FileSink> >::const_iterator itSinks = traceSink.persistantSinks.begin();
     itSinks != traceSink.persistantSinks.end(); ++itSinks) {
-    if ((*itSinks)->will_consume(set))
-      logging::core::get()->remove_sink(*itSinks);
+    if (itSinks->second->will_consume(set))
+      logging::core::get()->remove_sink(itSinks->second);
   }
   traceSink.persistantSinks.clear();
```

Comment:
Following our discussion, maybe we should remove this and juste remove the targeted sink?

Reply:
I think we should do so. It would be more logical, more coherent and more clear about what's going on in this method.